### PR TITLE
chore: shake Batteries

### PR DESCRIPTION
**Pull request by semorrison**

### CODE REVIEW

File: Batteries/Data/List/Perm.lean [hunk 1]
```diff
@@ -7,7 +7,6 @@ import Batteries.Tactic.Alias
 import Batteries.Data.List.Init.Attach
 import Batteries.Data.List.Pairwise
 -- Adaptation note: nightly-2024-03-18. We should be able to remove this after nightly-2024-03-19.
```
Comment (semorrison) [inline Batteries/Data/List/Perm.lean:9]:
```suggestion
```

File: Batteries/Data/Array/Basic.lean [hunk 2]
```diff
@@ -4,7 +4,6 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Arthur Paulino, Floris van Doorn, Jannis Limperg
 -/
 import Batteries.Data.List.Init.Attach
-import Batteries.Data.Array.Init.Lemmas
```
Comment (fgdorais) [inline Batteries/Data/Array/Basic.lean:7]: This one could stay. The import is currently empty but its purpose is to be included here (and ideally nowhere else).

File: Batteries/Data/List/Basic.lean [hunk 3]
```diff
@@ -3,7 +3,6 @@ Copyright (c) 2016 Microsoft Corporation. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Leonardo de Moura
 -/
-import Batteries.Data.List.Init.Lemmas
```
Comment (fgdorais) [inline Batteries/Data/List/Basic.lean:6]: This one could stay. The import is currently empty but its purpose is to be included here (and ideally nowhere else).

File: Batteries/Data/Array/Basic.lean (hunk 2 above)
Comment (semorrison) [reply Batteries/Data/Array/Basic.lean:7]: Sure... importing empty files for which there are no active plans to add anything seems a bit of a waste of time, but no objection.

File: Batteries/Data/List/Basic.lean (hunk 3 above)
Comment (semorrison) [reply Batteries/Data/List/Basic.lean]: Done.

**Review by fgdorais — APPROVED** [review]

**Pull request merged**